### ISSUE TITLE
Add option to zoom in sliderInput 

### ISSUE THREAD

**Issue by gogonzo** [insightsengineering/teal.slice#133]
Our sliderInput needs improvement for distributions where values are concentrated in a narrow range. It's hard to figure out where to set slider range. I propose option to zoom in the histogram by the user. If user selects a range on the distribution, little popup will show up with the distribution of the zoomed range. 

![image](https://user-images.githubusercontent.com/6959016/203741468-1a1da6ed-c556-4667-9634-a8bfe9ba0ebe.png)

**Comment by lcd2yyz** [comment]
It would also be very convenient for users to be able to enter the desired value range as text input.
Something like the numerical filter slider in RStudio SDE could be pretty nice.
It allows user to drag and drop to select on a range, or enter the value directly in the text box below.
![image](https://user-images.githubusercontent.com/78242757/204711512-241f938a-c5a3-45b4-8d9f-19d018c9c6d9.png)

**Comment by nikolas-burkoff** [comment]
Yes - I was thinking that being able to add the range _might_ be all we actually need @lcd2yyz  (i.e. behave like the slider in teal.goshawk - if you want to see the zoomed in distribution you can always use the variable browser to see what you are filtering) - it would then be much simpler both to use and to implement.

![image](https://user-images.githubusercontent.com/15201933/210327540-57c03409-af54-4d9d-b9fa-846b30ce5645.png)
![image](https://user-images.githubusercontent.com/15201933/210327578-d1277ee7-d2c1-48c2-80f0-a090bd6179eb.png)

and if this was a widget in teal.widgets it could be reused in teal.goshawk...

One thing we need to be careful with is at the moment it's not possible to create invalid filters in the ui of the filter panel, but by adding input boxes it will be possible - so we should add shinyvalidate (?) and handle the case where the filter panel filtering would produce an error carefully (e.g. not apply filter until it's valid, have a save filter button which is disabled until it's valid)

**Comment by gogonzo** [comment]
@nikolas-burkoff Looks like we agree on this https://github.com/insightsengineering/teal.goshawk/issues/205

**Comment by nikolas-burkoff** [comment]
See https://github.com/insightsengineering/teal.slice/pull/144

**Comment by nikolas-burkoff** [comment]
Note 
![image](https://user-images.githubusercontent.com/15201933/211526896-7ce4c1a8-16cc-42c3-86ac-7df133eb7ab9.png)

**Comment by nikolas-burkoff** [comment]
shinyvalidate plays nicely with the various inputs:


![Image](https://user-images.githubusercontent.com/15201933/214589048-d0b523f0-c819-42ff-9915-7de98eab12e0.png)

Though note for FilterStateDateTime we use 2 indpendent datepickers rather than a single range like in this example, which would can also be handled but you need the value of the "other" one included in the validation function



```
library(shiny)
library(shinyvalidate)
library(shinyWidgets)

ui <- fluidPage(
  numericRangeInput("range", "Select a range", value = c(10, 20)),
  dateRangeInput("date", "Select a date range", start = "2023-01-10", end = "2023-01-20"),
  airDatepickerInput("date2", "Select another date range", range = TRUE, value = c("2023-01-10", "2023-01-20")),
  airDatepickerInput("date_time", "Select a date range", range = TRUE, timepicker = TRUE, value = c("2023-01-10", "2023-01-20"))
)

server <- function(input, output, session) {
  iv <- shinyvalidate::InputValidator$new()

  iv$add_rule("range", sv_required("Invalid numeric range"))
  iv$add_rule("date", sv_required("Invalid date range"))
  iv$add_rule("date2", sv_required("Invalid date range"))
  iv$add_rule("datetime", sv_required("Invalid date time range"))
  iv$add_rule("range", ~ if ((.)[[1]] > (.)[[2]]) "Invalid numeric range")
  iv$add_rule("date", ~ if ((.)[[1]] > (.)[[2]]) "Invalid date range")
  iv$add_rule("date2", ~ if (length(.) < 2) "Invalid date range")
  iv$add_rule("date_time", ~ if (length(.) < 2) "Invalid datetime range")


  iv$enable()
}

shinyApp(ui, server)
```

**Comment by gogonzo** [comment]
Blocked to avoid conflicts with #129

**Comment by asbates** [comment]
@gogonzo I think this can be un-blocked now?

Also I'm not exactly sure what is needed here. 

- I think we don't want a zoom but rather 2 input boxes? 
- How much of #144 do we need? For example,
  - What is the reset button supposed to do?
  - Do we need to add the validation? This feels like another issue to me because then we should do validation for all inputs.

**Comment by nikolas-burkoff** [comment]
So I think the only inputs which need validation are this new one and the date/date time range pickers ( https://github.com/insightsengineering/teal.slice/issues/158) and those have been done

**Comment by gogonzo** [comment]
@asbates 
> - I think we don't want a zoom but rather 2 input boxes?

True, we prefer two input boxes

> - How much of https://github.com/insightsengineering/teal.slice/pull/144 do we need? For example,
>    - What is the reset button supposed to do?
>    - Do we need to add the validation? This feels like another issue to me because then we should do validation for all inputs.

- Reset is something for later - it will reset `selected` to `choices`. 
- Regarding validation - If I was doing this issue I'd probably do this if I felt it's missing. Some inputs already have validation (date and datetime).

**Comment by donyunardi** [comment]
This is blocked by:
* https://github.com/insightsengineering/teal.slice/issues/233
* https://github.com/insightsengineering/teal.slice/issues/234
